boards/arm/adafruit_kb2040/adafruit_kb2040_defconfig

```diff
@@ -0,0 +1,27 @@
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SOC_SERIES_RP2XXX=y
+CONFIG_SOC_RP2040=y
+CONFIG_BOARD_ADAFRUIT_KB2040=y
+
+CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC=125000000
+
+# Enable GPIO
+CONFIG_GPIO=y
```

Comment:
If the board requires GPIO in order to _boot_ and use the console (e.g. if it has a GPIO controlled regulator, which needs to be turned on), GPIO should be enabled by default. Otherwise not. And yet, we do have a lot of boards enabling GPIO by default where it is not needed.